### PR TITLE
Plumb additional TLS configuration

### PR DESCRIPTION
**Pull request by zalegrala**

### CODE REVIEW

**Comment by zalegrala** [comment]
Testing this out doesn't seem to work correct.  I have the following, but I don't think the new TLS values are being used.  Any pointers?
```js
const client = new tracing.Client({
  endpoint: 'https://ingest.trace.svc.cluster.znet:4317',
  exporter: tracing.EXPORTER_OTLP,
  insecure: false,
  tls_server_name: 'tempo.trace.svc.cluster.znet',
  tls_ca_file: '/tls/ca.crt',
  tls_cert_file: '/tls/tls.crt',
  tls_key_file: '/tls/tls.key',
  headers: { 'X-Scope-OrgID': randomOrg },
})
```

File: tracing.go [hunk 1]
```diff
@@ -146,6 +147,10 @@ type ClientConfig struct {
 	Exporter       exporterType `json:"type"`
 	Endpoint       string       `json:"url"`
 	Insecure       bool         `json:"insecure"`
+	TLSServerName  string       `json:"tls_server_name"`
+	TLSCertFile    string       `json:"tls_cert_file"`
+	TLSKeyFile     string       `json:"tls_key_file"`
+	TLSCAFile      string       `json:"tls_ca_file"`
```
Comment (mapno) [inline tracing.go:153]: WDYT of directly vendoring the otel TLS config block?

Comment (zalegrala) [reply tracing.go:153]: Ah yes, good idea.

**Review by mapno — APPROVED** [review]

**Comment by zalegrala** [comment]
I was able to validate these changes.  The `json` tags on the config object are not honored, and instead, the default translation is required.  This means that instead of the `json` tag of `server_name_override` on the go struct, the name of the struct is matched, and so just `server_name` is used.  Additionally, to specify the `CAFile` where the struct has `ca_file`, users need to specify `c_a_file`.

**Pull request merged**